### ISSUE TITLE
AssertionError parsing indexpage.xml, related to addition of mainpage?

### ISSUE THREAD

**Issue by bjfar** [mosra/m.css#37]
So now I am trying to add some extra pages to the documentation. For example I tried adding a \mainpage with this content:

```
/*!
 \mainpage The Main Page
 \section sec_1 Section 1
 ...section 1 stuff...
 \section sec_2 Section 2
 ... section 2 stuff...
*/
```

However the dox2html5.py script falls over when parsing indexpage.xml like this:

```
DEBUG:root:Parsing indexpage.xml
Traceback (most recent call last):
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 2895, in <module>
    run(doxyfile, os.path.abspath(args.templates), args.wildcard, args.index_pages, search_merge_subtrees=not args.search_no_subtree_merging, search_add_lookahead_barriers=not args.search_no_lookahead_barriers, search_merge_prefixes=not args.search_no_prefix_merging)
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 2813, in run
    parsed = parse_xml(state, file)
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 1938, in parse_xml
    compound.description, templates, compound.sections, footer_navigation, example_navigation, search_keywords, compound.is_deprecated = parse_toplevel_desc(state, compounddef.find('detaileddescription'))
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 1316, in parse_toplevel_desc
    parsed = parse_desc_internal(state, element)
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 651, in parse_desc_internal
    parsed = parse_desc_internal(state, i)
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 677, in parse_desc_internal
    id = extract_id_hash(state, element)
  File "/home/farmer/repos/gambit/myfork/m.css/doxygen/dox2html5.py", line 415, in extract_id_hash
    assert base_url == state.current_url
AssertionError
```

Raw doxygen seems to have no problem with it:

![doxygen](https://user-images.githubusercontent.com/2520133/39309522-267029d4-4960-11e8-963c-3af54440d107.png)

I looked in indexpage.xml, it doesn't look that weird to me, but perhaps you will see something :). I have attached both the full output and indexpage.xml (renamed to indexpage_xml.txt to make github happy). This is the same project as my last issue report so you can find everything at https://github.com/bjfar/gambit_1.1/tree/docs_pelican, in the 'doc' directory.

[mcss_out.txt](https://github.com/mosra/m.css/files/1951406/mcss_out.txt)
[indexpage_xml.txt](https://github.com/mosra/m.css/files/1951411/indexpage_xml.txt)

**Comment by mosra** [comment]
Argh, sorry, I'm to blame here. Doxygen names the sections `index_1sec_1` in the XML output, where the first `_1` is the section separator and the second `_1` is part of your section name... and my regex fails to match this properly, blowing up.

Temporary workaround: If you name the section *any other name* that doesn't have `_1` in it, it will work ;)

I'll push a proper fix for this in the next hours.

**Comment by mosra** [comment]
Should be fixed in 48e03c6be934cb2031e00c78db74b4e28eb161d4. Please reopen if it still appears.
